src/battle/battle_setup.c

```diff
@@ -0,0 +1,575 @@
+                SetMonData(pokemon, MON_DATA_EXPERIENCE, &exp);
+                CalcMonLevelAndStats(pokemon);
+            }
+            BattleSetup_AddMonToParty(setup, pokemon, BATTLER_PLAYER);
+        }
+        FreeToHeap(pokemon);
+    }
+
+    if (fieldSystem_unkA4 != NULL && sub_020290FC(fieldSystem_unkA4, 12)) {
+        setup->unk_194 = 1;
+    }
+    Save_Bag_Copy(bag, setup->bag);
+    Pokedex_Copy(pokedex, setup->pokedex);
+    Options_Copy(options, setup->options);
+    BattleSetup_SetChatotVoiceClip(setup, chatot, BATTLER_PLAYER);
+    setup->storagePC = SaveArray_PCStorage_Get(fieldSystem->saveData);
+    setup->timeOfDay = Field_GetTimeOfDay(fieldSystem);
+    setup->unk_10C = fieldSystem->unk94;
+    setup->unk1B8 = fieldSystem->unkB0;
+    setup->unk_12C = sub_0202CA44(fieldSystem->saveData);
+    setup->gameStats = Save_GameStats_Get(fieldSystem->saveData);
+    setup->mapNumber = fieldSystem->location->mapId;
+    setup->palPad = Save_PalPad_Get(fieldSystem->saveData);
+    setup->saveData = fieldSystem->saveData;
+
+    if (sub_0203401C(sub_0203993C())) {
+        int avatar = PlayerProfile_GetAvatar(profile);
+        int gender = PlayerProfile_GetTrainerGender(profile);
+        setup->trainer[BATTLER_PLAYER].trainerClass = sub_0205B46C(gender, avatar, 1);
+        CopyU16StringArray(setup->trainer[BATTLER_PLAYER].name, PlayerProfile_GetNamePtr(setup->profile[BATTLER_PLAYER]));
+        setup->trainer[BATTLER_PLAYER2] = setup->trainer[BATTLER_PLAYER];
+    } else {
+        sub_02052580(setup);
+    }
+}
+
+void sub_020522F0(BattleSetup* setup, FieldSystem *fieldSystem, void *a1) {
+    sub_020520B0(setup, fieldSystem, SaveArray_Party_Get(fieldSystem->saveData), a1);
+}
+
+static void sub_0205230C(FieldSystem* fieldSystem, PlayerProfile* profile1, PlayerProfile* profile2) {
+    SaveVarsFlags* vars_flags = Save_VarsFlags_Get(fieldSystem->saveData);
+    MomsSavings* savings = SaveData_GetMomsSavingsAddr(fieldSystem->saveData);
+
+    if (Save_VarsFlags_MomsSavingsFlagCheck(vars_flags)) {
+        u32 money2 = PlayerProfile_GetMoney(profile2);
+        int delta = (int)(money2 - PlayerProfile_GetMoney(profile1)) / 4;
+        u32 savingsBalance = MomSavingsBalanceAction(savings, MOMS_BALANCE_GET, 0);
+        u32 r1;
+        if (delta > 0) {
+            if (savingsBalance + delta >= 999999) {
+                delta = 999999 - savingsBalance;
+            }
+            PlayerProfile_SetMoney(profile2, money2 - delta);
+            r1 = MomSavingsBalanceAction(savings, MOMS_BALANCE_ADD, delta);
+        } else {
+            r1 = savingsBalance;
+        }
+        if (sub_0209322C(savings, r1, savingsBalance)) {
+            sub_02092E14(FieldSystem_GetGearPhoneRingManager(fieldSystem), 12, TRUE);
+        }
+    }
+}
+
+void sub_0205239C(BattleSetup* setup, FieldSystem* fieldSystem) {
+    PlayerProfile* profile;
+    Party* party;
+    Bag* bag;
+    Pokedex* pokedex;
+    SaveVarsFlags* vars_flags;
```

Comment:
change snake_case -> camelCase

Reply:
How about no?

Reply:
Fine I'll just do it in a later PR if I remember